### PR TITLE
refactor: simplifies Feature Manifests

### PR DESCRIPTION
**Pull request by bartoszmajsak**
## Description

The raw and template manifest structs differed only in their content processing methods, making it an unnecessary abstraction. This work consolidates the two implementations, reducing duplication and resulting in cleaner code.

The implementation still adheres to the convention of using a `.tmpl.` in the file name to indicate a template and only then process it as such.

The feature builder has been reworked from chaining `ManifestLocation().Manifest` in favor of composition. This not only makes the code easier to follow but also allows to potentially mix different kinds of Manifest providers.

Additionally, `Applier` wrapper has been introduced to handle different strategies of applying manifests to the cluster based on their types.

## How Has This Been Tested?

- create DSCI with Service Mesh enabled
- create DSC with KServe enabled
- observe the successful application of all features
  - inspect the status of DSCI and DSC
  - inspect the status of FeatureTrackers

## Merge criteria

- [x] You have read the [contributors guide](https://github.com/opendatahub-io/opendatahub-operator/blob/incubation/CONTRIBUTING.md).
- [x] Commit messages are meaningful - have a clear and concise summary and detailed explanation of what was changed and why.
- [x] Pull Request contains a description of the solution, a link to the JIRA issue, and to any dependent or related Pull Request.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: apis/features/v1/features_types.go [hunk 1]
```diff
@@ -58,9 +58,10 @@ var ConditionReason = struct {
 }
 
 const (
-	ComponentType OwnerType = "Component"
-	DSCIType      OwnerType = "DSCI"
-	UnknownType   OwnerType = "Unknown"
+	ComponentType          OwnerType = "Component"
+	DSCIType               OwnerType = "DSCI"
+	PlatformCapabilityType OwnerType = "Platform"
```
Comment (bartoszmajsak) [inline apis/features/v1/features_types.go:63]: This is not needed yet, I will remove

```suggestion
```

File: controllers/dscinitialization/servicemesh_setup.go [hunk 2]
```diff
@@ -195,6 +202,9 @@ func (r *DSCInitializationReconciler) authorizationFeatures(instance *dsciv1.DSC
 					servicemesh.EnsureServiceMeshInstalled,
 					servicemesh.EnsureAuthNamespaceExists,
 				).
+				PostConditions(
+					feature.WaitForPodsToBeReady(serviceMeshSpec.ControlPlane.Namespace),
+				).
```
Comment (bartoszmajsak) [inline controllers/dscinitialization/servicemesh_setup.go:207]: It seems it got lost in previous PR

https://github.com/opendatahub-io/opendatahub-operator/blob/odh-2.14.0/controllers/dscinitialization/servicemesh_setup.go#L195C5-L195C74

File: pkg/feature/builder.go [hunk 3]
```diff
@@ -56,51 +60,40 @@ func Define(featureName string) *featureBuilder {
...
 
 	return fb
 }
 
-// Used to enforce that Manifests() is called after ManifestsLocation() in the chain.
-type requiresManifestSourceBuilder struct {
-	*featureBuilder
-}
+// TargetNamespace sets the namespace in which the feature should be applied.
+// If not set, the feature will be applied in the application namespace (where this operator lives).
```
[the remark this answers is not in the document]
Comment (bartoszmajsak) [reply pkg/feature/builder.go:70]: I accidentally edited your comment above, but reverted to original... Apologies. It seems I have superpowers xD

Comment (bartoszmajsak) [inline pkg/feature/builder.go:70]:
```suggestion
// If not set, the feature will be applied in the application namespace.
```

That was clearly a hallucination. Thanks for spotting

File: pkg/feature/builder.go [hunk 4]
```diff
@@ -196,15 +189,16 @@ func (fb *featureBuilder) OnDelete(cleanups ...Action) *featureBuilder {
 // Create creates a new Feature instance and add it to corresponding FeaturesHandler.
 // The actual feature creation in the cluster is not performed here.
 func (fb *featureBuilder) Create() (*Feature, error) {
+	alwaysEnabled := func(_ context.Context, _ *Feature) (bool, error) {
```
[the remark this answers is not in the document]
Comment (bartoszmajsak) [reply pkg/feature/builder.go:192]: I am not sure what is the question about. Can you clarify?

I extracted the function for better readability only and made both parameters blank as they are not used.

**Review by zdtsw — APPROVED** [review]

**Comment by bartoszmajsak** [comment]
Thanks for taking care of it @zdtsw

**Review by zdtsw — COMMENTED** [review]
/lgtm

**Pull request merged**